osmtm/views/user.py

```diff
@@ -44,10 +44,15 @@ def users(request):
 def users_json(request):
     query = DBSession.query(User).order_by(User.username)
 
+    q = ''
     if 'q' in request.params:
-        q = request.params.get('q')
-        query = query.filter(User.username.ilike('%' + q + '%')).limit(10)
+        q = request.params.get('q').strip()
 
+    if q:
+        query = query.filter(User.username.ilike('%' + q + '%')).limit(10)
+    else:
+        request.response.cache_control = \
+            'private, max-age=600'
```

Comment:
Definitely; perhaps this could be expanded to other resources that are largish, yet change slowly. IIRC there's currently no cron to set up, but static serving would help immensely.

Note that the two approaches can be combined: `max-age` would cache the response for a few minutes *without any network requests* performed to the resource, after this timed out, an `etag` would return a 304 for a fresh resource.  (IMHO rather `Last-Modified` would be the right tool here instead of `ETag`, given the cron-regeneration, but those are otherwise functionally equivalent)